CoreUtilities/src/au/gov/asd/tac/constellation/utilities/rest/RestClient.java

```diff
@@ -114,29 +175,74 @@ public static URL generateUrl(final String url, final Map<String, String> params
      * HTTP response content.
      */
     protected byte[] bytes;
-
-    public abstract HttpsURLConnection makeGetConnection(final String url, final Map<String, String> params) throws IOException;
+    
+    /**
+     * Manage the creation of a HTTP GET connection.
+     * Deprecated - use makeGetConnection(final String url, final List<Tuple<String, String>> params)
+     * instead.
+     * 
+     * @param url The URL to base the GET connection on.
+     * @param params Any parameters to add to the URL
+     * @return HttpsURLConnection object corresponding to the created connection
+     * @throws IOException 
+     * @deprecated
+     */
+    @Deprecated
+    public HttpsURLConnection makeGetConnection(final String url, final Map<String, String> params) throws IOException {
+        throw new UnsupportedOperationException("This method is obsolete");
+    }
+    
+    /**
+     * Manage the creation of a HTTP GET connection.
+     * 
+     * @param url The URL to base the GET connection on.
+     * @param params Any parameters to add to the URL
+     * @return HttpsURLConnection object corresponding to the created connection
+     * @throws IOException 
+     */
+    public HttpsURLConnection makeGetConnection(final String url, final List<Tuple<String, String>> params) throws IOException {
+        throw new UnsupportedOperationException("This method must be overridden");
```

Comment:
could this be made abstract instead?

Reply:
not yet as it would break all old implementqations. Hence went for approach of deprecating old version, and adding new version. Once we are satisfied we have no old implementations we can remove old and make new abstract again.
Ie
If we make abstract all old implmentations immediately break during compile.

Reply:
Following our chat, Ive removed all the deprecated methods, which enables me to again make these methods abstract.